test-lib.sh

```diff
@@ -514,8 +514,10 @@ EOF
   local id_file
   mkdir -p "${APP_ID_FILE_DIR:?}"
   id_file="${APP_ID_FILE_DIR:?}"/"$RANDOM"
-  if ! docker build -f "$tmpdir/Dockerfile" --no-cache "$tmpdir" --iidfile "$id_file"; then
-    echo "  ERROR: Failed to find $binary in Dockerfile!" >&2
+  docker build -f "$tmpdir/Dockerfile" -q --no-cache "$tmpdir" > "$id_file"
```

Comment:
In case $id_file is not checked why the output is redirected to $id_file?
Any reason for this?

Reply:
What do you mean by "$id_file is not checked"?

Reply:
> Do we really need -q? What about the result of why the image build failed?

That goes to stderr. 
We need -q, because if it is not present then whole output of the build goes to "$id_file." Id did not find any other option for separating the image id, as the --id-file parameter can not be used.

Reply:
We do not see the output of `docker build` and we can not analyze troubles.

Reply:
And what if we check the return value and when it does not equal 0 we run the build again with all log going to stdout?

Reply:
That could be possible only if there is some equivalent parameter to `--iidfile` in the docker which is available in rhel7.
However there is no equivalent. So I do not know how the suggested approach could be implemented.

Another possibility could be to write some build function, which will build image with `--iidfile` if possible, and `-q` parameter otherwise.

Reply:
> And what if we check the return value and when it does not equal 0 we run the build again with all log going to stdout?

Do not run the same command twice.

Reply:
> That could be possible only if there is some equivalent parameter to `--iidfile` in the docker which is available in rhel7. However there is no equivalent. So I do not know how the suggested approach could be implemented.
> 
> Another possibility could be to write some build function, which will build image with `--iidfile` if possible, and `-q` parameter otherwise.

What about this approach:
```bash

  DOCKER_BUILD_OPTIONS="" # Option for docker command
  docker info 2>/dev/null | grep podman 1>/dev/null && is_podman=1 || is_podman=0
  if [ $is_podman -eq 1 ]; then
    DOCKER_BUILD_OPTIONS='--iidfile="$id_file"'
  fi

```

Based on the documentation https://docs.docker.com/engine/reference/commandline/build/
docker build has an option --iidfile.

Does it make sense to for ask to docker maintainer to add this option also to RHEL7?

Reply:
It would mean that cleaning app images for docker will not work. Are you OK with that?

Reply:
@phracek?

Reply:
@hhorak @pkubatrh Please could you look at this thread and state your opinion about this? I would like to have this fixed.

In my opinion the best approach would be to build quietly and the repeat the build command for logging purposes if the build fails.

Reply:
From the solutions discussed here I like see the `-q` more, also would like to avoid splitting behavior if possible. However, isn't this the same issue as we have in `build.sh` already? We might use the same (maybe simplified) approach: https://github.com/sclorg/container-common-scripts/blob/master/build.sh#L180

Reply:
Good idea, thanks.

Reply:
@hhorak this is now reworked. Please take a look.